### PR TITLE
Add config identity_min_length and apply non-admin validation

### PR DESCRIPTION
**Pull request by manfre**
Implements #8 

* Add setting `identity_min_length` to apply to non-admins
* Add test for CreateIdentity form
* Apply validation for existing settings `restricted_usernames`, `identity_max_per_user`

### CODE REVIEW

File: users/views/identity.py [hunk 1]
```diff
@@ -167,6 +179,12 @@ def clean(self):
             ):
                 raise forms.ValidationError(f"{username}@{domain} is already taken")
 
+            if not self.user.admin and (
+                Identity.objects.filter(users=self.user).count()
+                >= Config.system.identity_max_per_user
+            ):
+                raise forms.ValidationError("User may not have anymore identities")
```
Comment (andrewgodwin) [inline users/views/identity.py:186]: Let's phrase this as "You are not allowed more than {num} identities."

File: users/views/identity.py [hunk 2]
```diff
@@ -144,10 +145,21 @@ def __init__(self, user, *args, **kwargs):
...
             value = self.cleaned_data["username"].lstrip("@").lower()
+
+            if not self.user.admin:
+                # Apply username min length
+                limit = int(Config.system.identity_min_length)
+                validators.MinLengthValidator(limit)(value)
+
+                # Apply username restrictions
+                if value in Config.system.restricted_usernames.split():
+                    raise forms.ValidationError("Username not allowed.")
```
Comment (andrewgodwin) [inline users/views/identity.py:161]: Can we phrase this as "This username is restricted to administrators only."?

**Review by andrewgodwin — COMMENTED** [review]
Looking good - I really appreciate the tests! Just a few wording changes and I'll bring it in.

**Comment by manfre** [comment]
Changed error messages as requested. Squashed and rebased for merge.

**Comment by manfre** [comment]
Added `signed-off-by`

**Review by andrewgodwin — APPROVED** [review]

**Pull request merged**